backend/routes.js

```diff
@@ -474,6 +481,37 @@ export default class Servers {
             });
         });
 
+        /* Plotly v2 API requests */
+
+        server.post('/datacache', function getDatacacheHandler(req, res, next) {
+
+            const {payload, type} = req.params;
+            
+            if (type !== 'csv') {
+                const datacacheResp = newDatacache(payload, type);
+                datacacheResp.then(plotlyRes => plotlyRes.json().then(resJSON => {
+                    return res.json(plotlyRes.status, resJSON);
+                })).catch(err => {
+                    console.error(err);
+                    throw new Error(err);
+                });                
+            }
+            else {
+                const rand = Math.round(Math.random()*1000).toString();
+                const path = os.homedir().concat('/data_export_', rand, '.csv');
```

Comment:
this will break on windows machines since their `/` is really `\`. Instead, node's ``path.join` should work: 

```
path.join(os.homedir(), `data_export_${rand}.csv`)
```

Reply:
however, the user is guarenteed to have a folder at ~/.plotly/connector`, which is where we save a bunch of stuff, so we should probably put it there instead of littering their home directory.
```
downloadPath = path.join(getSetting('STORAGE_PATH'), `data_export_${rand}.csv`)
```